core/src/mindustry/logic/LStatements.java

```diff
@@ -2149,7 +2149,8 @@ void rebuild(Table table){
             table.button(b -> {
                 b.image(Icon.pencilSmall);
                 
-                b.clicked(() -> showSelect(b, GlobalVars.soundNames.toArray(String.class), id.substring(4), t -> {
+                String soundName = id.startsWith("@sfx-") ? id.substring(5) : id;
```

Comment:
Not sure if `id` should be replaced by an empty string when it's not a built-in sound effect.